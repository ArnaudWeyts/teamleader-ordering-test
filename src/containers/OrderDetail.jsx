import React, { Component } from 'react';
import { connect } from 'react-redux';

import { selectOrder, removeProductFromOrder } from '../actions/ordersActions';

import ProductList from '../components/ProductList';

class OrderDetail extends Component {
  componentDidMount() {
    const {
      dispatch,
      match: {
        params: { id }
      }
    } = this.props;

    dispatch(selectOrder(id));
  }

  render() {
    const { dispatch, order } = this.props;

    if (!order) {
      return <div>Loading...</div>;
    }

    return (
      <div>
        <h2>Order #{order.id}</h2>
        <h2>Items</h2>
        <ProductList
          items={order.items}
          removeProduct={(id, quantity) =>
            dispatch(removeProductFromOrder(id, quantity))
          }
        />
        <button type="button" onClick={() => console.log('add item')}>
          Add an item
        </button>
        <h2>Total: {order.total}</h2>
        <button type="button" onClick={() => console.log('order placed')}>
          Place order
        </button>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  order: state.orders.selectedOrder,
  products: state.orders.selectedOrderProducts
});

export default connect(mapStateToProps)(OrderDetail);
