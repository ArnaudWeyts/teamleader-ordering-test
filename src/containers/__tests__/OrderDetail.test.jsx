import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { MemoryRouter } from 'react-router-dom';

import OrderDetail from '../OrderDetail';

const mockStore = configureStore([thunk]);

it('renders without crashing', () => {
  const store = mockStore({ orders: { selectedOrder: null } });
  shallow(<OrderDetail store={store} />);
});

it('renders the order detail page loading', () => {
  const store = mockStore({
    orders: {
      allOrders: [],
      selectedOrder: null
    }
  });

  const props = {
    match: {
      params: {
        id: '1'
      }
    }
  };

  const component = renderer.create(
    <MemoryRouter initialEntries={['/']}>
      <OrderDetail {...props} store={store} />
    </MemoryRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the order detail page', () => {
  const store = mockStore({
    orders: {
      allOrders: [
        {
          id: '1',
          'customer-id': '1',
          items: [
            {
              'product-id': 'B102',
              quantity: 10,
              'unit-price': 4.99,
              total: 49.9
            }
          ],
          total: 49.9
        }
      ],
      selectedOrder: {
        id: '1',
        'customer-id': '1',
        items: [
          {
            'product-id': 'B102',
            quantity: 10,
            'unit-price': 4.99,
            total: 49.9
          }
        ],
        total: 49.9
      }
    }
  });

  const props = {
    match: {
      params: {
        id: '1'
      }
    }
  };

  const component = renderer.create(
    <MemoryRouter initialEntries={['/']}>
      <OrderDetail {...props} store={store} />
    </MemoryRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
