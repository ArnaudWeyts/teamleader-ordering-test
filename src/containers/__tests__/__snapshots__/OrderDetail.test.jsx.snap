// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the order detail page 1`] = `
<div>
  <h2>
    Order #
    1
  </h2>
  <h2>
    Items
  </h2>
  <div
    style={
      Object {
        "maxHeight": "70vh",
        "overflowY": "scroll",
      }
    }
  >
    <div
      className="ant-card ant-card-bordered"
    >
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <div
          className="ant-skeleton ant-skeleton-active"
        >
          <div
            className="ant-skeleton-content"
          >
            <h3
              className="ant-skeleton-title"
              style={
                Object {
                  "width": "38%",
                }
              }
            />
            <ul
              className="ant-skeleton-paragraph"
              style={undefined}
            >
              <li
                style={
                  Object {
                    "width": undefined,
                  }
                }
              />
              <li
                style={
                  Object {
                    "width": undefined,
                  }
                }
              />
              <li
                style={
                  Object {
                    "width": "61%",
                  }
                }
              />
            </ul>
          </div>
        </div>
        <h3>
          Quantity: 
          10
        </h3>
        <h3>
          Subtotal: 
          €49.90
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="ant-input-number"
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            onMouseOut={undefined}
            onMouseOver={undefined}
            style={Object {}}
          >
            <div
              className="ant-input-number-handler-wrap"
            >
              <span
                aria-disabled={false}
                aria-label="Increase Value"
                className="ant-input-number-handler ant-input-number-handler-up "
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                unselectable="unselectable"
              >
                <i
                  className="anticon anticon-up ant-input-number-handler-up-inner"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="up"
                    fill="currentColor"
                    height="1em"
                    onClick={undefined}
                    style={undefined}
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 0 0 140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                    />
                  </svg>
                </i>
              </span>
              <span
                aria-disabled={true}
                aria-label="Decrease Value"
                className="ant-input-number-handler ant-input-number-handler-down ant-input-number-handler-down-disabled"
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                role="button"
                unselectable="unselectable"
              >
                <i
                  className="anticon anticon-down ant-input-number-handler-down-inner"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="down"
                    fill="currentColor"
                    height="1em"
                    onClick={undefined}
                    style={undefined}
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </i>
              </span>
            </div>
            <div
              aria-valuemax={10}
              aria-valuemin={1}
              aria-valuenow={1}
              className="ant-input-number-input-wrap"
              role="spinbutton"
            >
              <input
                autoComplete="off"
                autoFocus={undefined}
                className="ant-input-number-input"
                disabled={undefined}
                id={undefined}
                max={10}
                maxLength={undefined}
                min={1}
                name="removeQuantity"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={undefined}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseUp={[Function]}
                pattern={undefined}
                placeholder={undefined}
                readOnly={undefined}
                required={false}
                step={1}
                tabIndex={undefined}
                type={undefined}
                value="1"
              />
            </div>
          </div>
          <button
            className="ant-btn ant-btn-default"
            onClick={[Function]}
            type="submit"
          >
            <span>
              Remove product(s)
            </span>
          </button>
        </form>
      </div>
    </div>
  </div>
  <div>
    <button
      className="ant-btn ant-btn-default"
      onClick={[Function]}
      type="button"
    >
      <span>
        Add an item
      </span>
    </button>
  </div>
  <h2>
    Total: 
    €49.90
  </h2>
  <button
    className="ant-btn ant-btn-primary"
    onClick={[Function]}
    type="button"
  >
    <span>
      Place order
    </span>
  </button>
</div>
`;

exports[`renders the order detail page loading 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <div
    className="ant-spin ant-spin-spinning"
  >
    <span
      className="ant-spin-dot ant-spin-dot-spin"
    >
      <i />
      <i />
      <i />
      <i />
    </span>
  </div>
</div>
`;
